tests/scripts/all.sh

```diff
@@ -957,6 +949,20 @@ component_build_no_sockets () {
     make CC=gcc CFLAGS='-Werror -Wall -Wextra -O0 -std=c99 -pedantic' lib
 }
 
+component_test_memory_buffer_allocator () {
+    msg "build: default config with memory buffer allocator enabled"
+    scripts/config.pl set MBEDTLS_MEMORY_BUFFER_ALLOC_C
+    scripts/config.pl set MBEDTLS_MEMORY_BACKTRACE
+    CC=gcc cmake -D CMAKE_BUILD_TYPE:String=Asan .
```

Comment:
Is Asan useful with the buffer allocator? Is running `ssl-opt.sh` useful? Does anybody even use it?

Reply:
Running `ssl-opt.sh` is useful I think, because `ssl_client2` and `ssl_server2` use the buffer allocator, too, so we test it with some 'real world' alloc/free data.